src/csharp/Microsoft.Spark/Sql/CastUnpickledItems.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.Spark.Sql
+{
+    /// <summary>
+    /// Cast UnpickledItems.
+    /// </summary>
+    internal class CastUnpickledItems
+    {
+        /// <summary>
+        /// Cast UnpickledItems from ArrayList to Array.
+        /// </summary>
+        /// <param name="unpickledItems">Unpickled items</param>
+        /// <returns>Cast unpickled items</returns>
+        public static object[] CastToArray(object unpickledItems)
+        {
+            var castUnpickledItems = new List<object>();
+            foreach (object[] objArr in (object[])unpickledItems)
+            {
+                var castObjArr = new List<object>();
+                ArrayList arrli = (ArrayList)objArr[0];
```

Comment:
Let's rename `arrli` to maybe `arrList` or something similar?

Reply:
Sounds good, changed.